// @atproto/api/dist/client/types/com/atproto/sync/subscribeRepos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isCommit = isCommit;
exports.validateCommit = validateCommit;
exports.isIdentity = isIdentity;
exports.validateIdentity = validateIdentity;
exports.isAccount = isAccount;
exports.validateAccount = validateAccount;
exports.isHandle = isHandle;
exports.validateHandle = validateHandle;
exports.isMigrate = isMigrate;
exports.validateMigrate = validateMigrate;
exports.isTombstone = isTombstone;
exports.validateTombstone = validateTombstone;
exports.isInfo = isInfo;
exports.validateInfo = validateInfo;
exports.isRepoOp = isRepoOp;
exports.validateRepoOp = validateRepoOp;
const lexicons_1 = require("../../../../lexicons");
const util_1 = require("../../../../util");
const is$typed = util_1.is$typed, validate = lexicons_1.validate;
const id = 'com.atproto.sync.subscribeRepos';
const hashCommit = 'commit';
function isCommit(v) {
    return is$typed(v, id, hashCommit);
}
function validateCommit(v) {
    return validate(v, id, hashCommit);
}
const hashIdentity = 'identity';
function isIdentity(v) {
    return is$typed(v, id, hashIdentity);
}
function validateIdentity(v) {
    return validate(v, id, hashIdentity);
}
const hashAccount = 'account';
function isAccount(v) {
    return is$typed(v, id, hashAccount);
}
function validateAccount(v) {
    return validate(v, id, hashAccount);
}
const hashHandle = 'handle';
function isHandle(v) {
    return is$typed(v, id, hashHandle);
}
function validateHandle(v) {
    return validate(v, id, hashHandle);
}
const hashMigrate = 'migrate';
function isMigrate(v) {
    return is$typed(v, id, hashMigrate);
}
function validateMigrate(v) {
    return validate(v, id, hashMigrate);
}
const hashTombstone = 'tombstone';
function isTombstone(v) {
    return is$typed(v, id, hashTombstone);
}
function validateTombstone(v) {
    return validate(v, id, hashTombstone);
}
const hashInfo = 'info';
function isInfo(v) {
    return is$typed(v, id, hashInfo);
}
function validateInfo(v) {
    return validate(v, id, hashInfo);
}
const hashRepoOp = 'repoOp';
function isRepoOp(v) {
    return is$typed(v, id, hashRepoOp);
}
function validateRepoOp(v) {
    return validate(v, id, hashRepoOp);
}
